### PR TITLE
[Refinery] Create URI to string transformation, to get the origin string to create e.g. a href element from the URI object

### PR DESCRIPTION
**Pull request by legionth**
This PR as proposal to create a transformation to transform an `Data\URI` object into a `string`.

The original idea was born during the development of #2042 , because there was no actual functionality to pass an `Data\URI` object around and be able to get the origin entered string back

Tell me what you think :+1:

### CODE REVIEW

**Comment by mjansenDatabay** [comment]
Hi @legionth,

I highly apprecitate this. Putting this to a `Transformation` is IMO the superior approach compared to a  `UrlToStringBuilder` (or something similar) in `./src/Data`, `./src/HTTP`  or `./Services/Http`.

Kind regards,
@mjansenDatabay

**Review by mjansenDatabay — APPROVED** [review]

**Review by mjansenDatabay — CHANGES_REQUESTED** [review]
Hi @legionth,

I highly appreciate this. Putting this to a `Transformation` is IMO the superior approach compared to a  `UrlToStringBuilder` (or something similar) in `./src/Data`, `./src/HTTP`  or `./Services/Http`.

Please provide a section in the README file.

Kind regards,
@mjansenDatabay

**Review by klees — APPROVED** [review]
Hi @legionth,

I'm fine with this too. Would you introduce the according factories as well and wire them up in the main factory of the refinery? I would suggest to call the `Transformation` `toString` then, this will lead to `$factory->uri()->toString()`.

Thanks!!

File: src/Refinery/URI/Group.php [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+/**
+ * @author  Niels Theen <ntheen@databay.de>
+ */
+
+namespace ILIAS\Refinery\URI;
+
+class Group
+{
+    public function string()
```
Comment (mjansenDatabay) [inline src/Refinery/URI/Group.php]: According to @klees this should be name `toString()`

**Comment by mjansenDatabay** [comment]
Thx @legionth for the additions. Please use `declare(strict_types=1);` and return type declarations where possible (e.g. in the factories and unit tests).

**Review by mjansenDatabay — APPROVED** [review]

**Pull request merged**